core/logic/src/main/java/org/apache/syncope/core/logic/LoggerLogic.java

```diff
@@ -216,10 +228,71 @@ private LoggerTO setLevel(final String name, final Level level, final LoggerType
         syncopeLogger.setLevel(LoggerLevel.fromLevel(level));
         syncopeLogger = loggerDAO.save(syncopeLogger);
 
+        boolean isAudit = LoggerType.AUDIT.equals(syncopeLogger.getType());
         LoggerContext ctx = (LoggerContext) LogManager.getContext(false);
+        String domainAuditLoggerName =
+                AuditManagerImpl.getDomainAuditEventLoggerName(AuthContextUtils.getDomain(), syncopeLogger.
+                        getKey());
         LoggerConfig logConf = SyncopeConstants.ROOT_LOGGER.equals(name)
                 ? ctx.getConfiguration().getLoggerConfig(LogManager.ROOT_LOGGER_NAME)
-                : ctx.getConfiguration().getLoggerConfig(name);
+                : isAudit
+                        ? ctx.getConfiguration().getLoggerConfig(domainAuditLoggerName)
+                        : ctx.getConfiguration().getLoggerConfig(name);
+
+        if (isAudit) {
+            // SYNCOPE-1144 For each custom audit appender class add related appenders to log4j logger
```

Comment:
Isn't this code a copy of what inserted in `LoggerLoader`? If so, please extract it into a public method in `LoggerLoader` and invoke that from here too.